packages/artifact/src/internal/download-http-client.ts

```diff
@@ -219,6 +219,13 @@ export class DownloadHttpClient {
       fileDownloadPath: string
     ): Promise<void> => {
       destinationStream.close()
+      // await until file is created at downloadpath; node15 and up fs.createWriteStream had not created a file yet
```

Comment:
The single code change we need in node16: awaiting `destinationStream.on('close',` ensures that the file has been created so we can `rmFile` delete it on line 229.

Reply:
❤️🙏